src/stashcache.py

```diff
@@ -543,20 +543,38 @@ def _namespace_dict(ns: Namespace):
         }
 
         for cache_name, cache_resource_obj in cache_resource_objs.items():
-            if resource_allows_namespace(cache_resource_obj, ns) and namespace_allows_cache_resource(ns, cache_resource_obj):
+            if resource_allows_namespace(cache_resource_obj, ns) and namespace_allows_cache_resource(ns,
+                                                                                                     cache_resource_obj):
                 nsdict["caches"].append(cache_resource_dicts[cache_name])
         return nsdict
+
+    def _resource_has_downed_cache(r: Resource, t: Topology):
+        if r.name not in t.present_downtimes_by_resource:
+            return False
+        downtimes = t.present_downtimes_by_resource[r.name]
+        for dt in downtimes:
+            try:
+                if XROOTD_CACHE_SERVER in dt.service_names:
+                    return True
+            except (KeyError, AttributeError):
+                continue
```

Comment:
Too bad we don't have a filter for this :(

Reply:
A lot of these queries would be soooo much easier if I could do SQL queries on Topology...